dashboard/src/components/Tabs/Filters.tsx

```diff
@@ -0,0 +1,215 @@
+import type { Dispatch, SetStateAction } from 'react';
+import { memo, useCallback, useMemo } from 'react';
+import { useIntl } from 'react-intl';
+
+import type {
+  ICheckboxSection,
+  Section,
+} from '@/components/Filter/CheckboxSection';
+import FilterCheckboxSection from '@/components/Filter/CheckboxSection';
+
+import FilterTimeRangeSection from '@/components/Filter/TimeRangeSection';
+
+import { DrawerSection } from '../Filter/Drawer';
+
+import type { RecordDiffType } from './tabsUtils';
+
+export const mapFilterToReq = <T extends Record<string, string>>(
+  filter: T,
+  fieldMap: Record<string, string>,
+): { [key: string]: string[] } => {
+  const filterMapped: { [key: string]: string[] } = {};
+
+  Object.entries(fieldMap).forEach(([reqField, field]) => {
+    const values = filter[field];
+    if (!values) return;
+
+    if (typeof values === 'object') {
+      Object.entries(values).forEach(([value, isSelected]) => {
+        if (isSelected) {
+          if (reqField === 'treeDetails.valid') {
+            value = value === 'Success' ? 'true' : 'false';
+          }
+          if (!filterMapped[reqField]) {
+            filterMapped[reqField] = [];
+          }
+          filterMapped[reqField].push(value);
+        }
+      });
+    } else {
+      filterMapped[reqField] = [values.toString()];
+    }
+  });
+  return filterMapped;
+};
+
+const parseCheckboxFilter = <T extends Record<string, Record<string, boolean>>>(
+  filter: T,
+  diffFilter: T,
+  isTFilterObjectKeys: (key: string) => boolean,
+): T => {
+  const result = structuredClone(filter) as T;
+  Object.keys(result).forEach(key => {
+    const currentFilterSection = result[key];
+
+    if (!currentFilterSection || !isTFilterObjectKeys(key)) {
+      return;
+    }
+
+    const diffFilterSection = diffFilter[key];
+
+    if (diffFilterSection) {
+      Object.keys(diffFilterSection).forEach(filterSectionKey => {
+        currentFilterSection[filterSectionKey] =
+          diffFilterSection[filterSectionKey];
+      });
+    }
+  });
+
+  return result;
+};
+
+const changeCheckboxFilterValue = <
+  T extends Record<string, Record<string, boolean>>,
+  K extends keyof T,
+>(
+  filter: T,
+  filterField: K,
+  value: string,
+): T => {
+  const newFilter: T = JSON.parse(JSON.stringify(filter ?? {}));
+  if (!newFilter[filterField]) {
+    newFilter[filterField] = {} as T[K];
+  }
+
+  const filterSection = newFilter[filterField] as Record<string, boolean>;
+  const filterValue = filterSection[value] ?? false;
+  filterSection[value] = !filterValue;
+
+  return newFilter;
+};
+
+type SectionsProps<T extends RecordDiffType> = {
+  diffFilter: T;
+  setDiffFilter: Dispatch<SetStateAction<T>>;
+};
+
+interface ICheckboxSectionProps<
+  T extends Record<string, Record<string, boolean>>,
+> extends SectionsProps<T> {
+  filter: T;
+  isTFilterObjectKeys: (key: string) => boolean;
+  sections: Section[];
+  //getSections: (parsedFilter: T) => ICheckboxSection[];
```

Comment:
this Section type is the new interface you created, and the line below it uses ICheckboxSection. Please check the need for the interface and the need for this prop that you left in a comment